roofit/roofitcore/inc/RooArgSet.h

```diff
@@ -75,35 +75,75 @@ class RooArgSet : public RooAbsCollection {
 	    const RooAbsArg& var5, const RooAbsArg& var6, 
 	    const RooAbsArg& var7, const RooAbsArg& var8, 
 	    const RooAbsArg& var9, const char *name="");
+  /// Create from any kind of container that can be iterated and holds pointers
+  /// to a type that derives from RooAbsArg.
+  template <class Input_t,
+      class Container_t = typename std::remove_reference<Input_t>::type,
+      class = typename std::enable_if<std::is_convertible<typename Container_t::value_type, const RooAbsArg*>::value>::type >
+  RooArgSet(Input_t inputs, const char* name = "") :
+  RooAbsCollection(name)
+  {
+    for (const auto elm : inputs) {
+      add(elm);
+    }
+  }
+  /// Create a non-owning collection from an initializer_list that holds
+  /// pointers to a type that derives from RooAbsArg.
+  /// `RooAbsArg({x, y, z})`
+  template<typename T,
+      typename = typename std::enable_if<std::is_convertible<T, const RooAbsArg*>::value>::type >
+  RooArgSet(std::initializer_list<const T> inputs, const char* name = "") :
+  RooAbsCollection(name)
+  {
+    for (const auto elm : inputs) {
+      add(elm);
+    }
+  }
+
 
   virtual ~RooArgSet();
-  // Create a copy of an existing list. New variables cannot be added
-  // to a copied list. The variables in the copied list are independent
-  // of the original variables.
+  /// Create a copy of an existing list. New variables cannot be added
+  /// to a copied list. The variables in the copied list are independent
+  /// of the original variables.
   RooArgSet(const RooArgSet& other, const char *name="");
   virtual TObject* clone(const char* newname) const { return new RooArgSet(*this,newname); }
   virtual TObject* create(const char* newname) const { return new RooArgSet(newname); }
   RooArgSet& operator=(const RooArgSet& other) { RooAbsCollection::operator=(other) ; return *this ;}
 
-  virtual Bool_t add(const RooAbsCollection& col, Bool_t silent=kFALSE) {
-    // Add all elements in list to collection
-    return RooAbsCollection::add(col, silent);
-  }
-  virtual Bool_t addOwned(const RooAbsCollection& col, Bool_t silent=kFALSE) {
-    // Add all elements in list as owned components to collection
-    return RooAbsCollection::addOwned(col, silent);
-  }
-  virtual void addClone(const RooAbsCollection& col, Bool_t silent=kFALSE) {
-    // Add owned clone of all elements of list to collection
-    RooAbsCollection::addClone(col, silent);
-  }
+  using RooAbsCollection::add;
+  virtual Bool_t add(const RooAbsArg& var, Bool_t silent=kFALSE);
```

Comment:
Where's the implementation now? Why even declare this if all it does is call the base class's `add`? (Ditto for `addOwned`. `addClone`.)

Reply:
It doesn't call the base class implementation. The derived actually needs to override, but the overloads that just dispatch to the "main" implementations were overridden as well. These overrides now have been removed, and the base-class implementations are now made visible using `using`.